### PR TITLE
Optional git commit message template

### PR DESCRIPTION
**Pull request by left23**
The change to the git config has to be done manually by each user. This is a security feature.

Use the commit message template as a starting point, and adjust to fit your requirements.

### Per repo
To use the git config in this repo, from the directory root, run: `git config --local include.path .gitconfig`
This will add:
```
[include]
	path = ../.gitconfig
```
to the `.git/config` file. The `.gitconfig` file defines the commit message template to use.

### Globally
To use globally, add the `.gitmessage` file (to the home directory, in this case) and run: `git config --global commit.template ~/.gitmessage`

### CODE REVIEW

**Comment by left23** [comment]
@lazysoundsystem Let me know if you think this will be useful. I'll use it personally, not sure if it has value in repos or not.

**Comment by lazysoundsystem** [comment]
Not sure if it's worth adding to each repo, but definitely in favor of a standard template we can refer to.
 
Here I've added prompting comments (which are ignored in the commit message itself):
```
# Form: <type>[optional scope]: <description>
# <type> is one of fix, feat, BREAKING CHANGE, build, chore, ci, docs, style, refactor, perf, test,
# [optional scope] names the part being changed
# <description> finishes the sentence "If applied, this commit will..."

# Leave a blank line

# Add Jira ticket IDs in the form PROPERTY-ticket_number, e.g. OPS-123
Refs: 
# Optional longer description goes here, after another blank line
```
I'm not sure if that list of types is appropriate for us (I took it from https://www.conventionalcommits.org/en/v1.0.0/ which says the list is based on the angular convention) - and I've noticed almost all of my conventional commits so far have been `chore: ...` so perhaps I'm using them wrong...

**Comment by lazysoundsystem** [comment]
If we put some documentation for the 'type' elsewhere, we could slim this down to:

```
# <conventional-commit-type>: If applied, this commit will...

# Leave a blank line

# Add Jira ticket IDs
Refs: 
# Optional longer description goes here, after another blank line
```
Comments at the end of the template could add more information, if it would be helpful.

File: .gitmessage [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<type>[optional scope]: <description>
+
+[optional body]
+
+Refs: ID-123
```
Comment (lazysoundsystem) [inline .gitmessage:5]: The extra keystrokes to delete these prompts might become annoying, the placeholders could be mentioned in comments instead.

Also, I think the way I've seen the optional scope used, it has another set of parentheses - e.g. `chore(tests): add test for...`

**Review by lazysoundsystem — APPROVED** [review]
No objections to including this - not sure if the comment about how to include it per repo is clear (the extra (`../` in the path) - perhaps something could be added to the README.

**Comment by left23** [comment]
> `# <conventional-commit-type>: If applied, this commit will...`

@lazysoundsystem I'm not sure what you mean by this "If applied, this commit will...". Can you elaborate and maybe provide an example?

**Comment by left23** [comment]
@lazysoundsystem I've made some adjustments. AFAIK, the Refs should be the last line, so I've swapped things around to accommodate. And everything except the Refs: is in comments, and I've added the link to the Conventional Commits docs, which I have relied on having bookmarked.

Let me know if you have more feedback.

I will look at clarifying the instructions now too.

**Comment by left23** [comment]
@attiks Any opinions on this? I know you've been using it for a while so probably don't need a git message template. Does it follow what you've been doing? Any suggestions to improve?

**Review by lazysoundsystem — APPROVED** [review]
Looks good to me!

**Pull request merged**